File: Algorithms/LCS.py
def LCS_Length(X, Y):
    m = len(X)
    n = len(Y)
    c = [[None]*(n+1) for i in range(m+1)]
    b = [[None]*(n+1) for i in range(m+1)]
    for i in range(m+1):
        for j in range(n+1):
            if i==0 or j==0:
                    c[i][j] = 0
            elif X[i-1] == Y[j-1]:
                c[i][j] = c[i-1][j-1]+1
                b[i][j] = 0
            elif c[i-1][j]>=c[i][j-1]:
                c[i][j] = c[i-1][j]
                b[i][j] = 1
            else:
                c[i][j] = c[i][j-1]
                b[i][j] = 2
    return c[m][n], b

def PrintLCS(b, X, i, j):
    if i==0 or j==0:
        return
    if b[i][j] == 0:
        PrintLCS(b, X, i-1, j-1)
        print(X[i-1], end='')
    elif b[i][j] == 1:
        PrintLCS(b, X, i-1, j)
    else:
        PrintLCS(b, X, i, j-1)

X = "ACCGGTCGAGTGCGCGGAAGCCGGCCGAA"
Y = "GTCGTTCGGAATGCCGTTGCTCTGTAAA"
# print(LCS_Length(X, Y))
i, b = LCS_Length(X, Y)
PrintLCS(b, X, len(X), len(Y))

File: Algorithms/LCS.cpp
#include <bits/stdc++.h>
using namespace std;

int LCS_Length(char* X, char* Y){
    int m = strlen(X);
    int n = strlen(Y);
    int c[m+1][n+1];
    for(int i=1; i<m+1; i++)
        c[i][0] = 0;
    for(int j=1; j<n+1; j++)
        c[0][j] = 0;
    for(int i=1; i<m+1; i++){
        for(int j=1; j<n+1; j++){
            if(X[i+1]==Y[j+1])
                c[i][j] = c[i-1][j-1]+1;
            else
                c[i][j] = max(c[i-1][j], c[i][j-1]);            
        }
    }
    return c[m][n];
}


int main(int argc, char const *argv[])
{
    char X[] = "ACCGGTCGAGTGCGCGGAAGCCGGCCGAA"; 
    char Y[] = "GTCGTTCGGAATGCCGTTGCTCTGTAAA"; 
      
    cout << "Length of LCS is " 
         << LCS_Length( X, Y)<<endl;
    return 0;
}
